src/main/scala/org/constellation/p2p/PeerAPI.scala

```diff
@@ -143,43 +145,109 @@ class PeerAPI(override val ipManager: IPManager[IO])(
+                  )
+                  val getInfo = dao.snapshotService.getSnapshotInfo.flatMap { info =>
+                    val checkpointsToGet = info.acceptedCBSinceSnapshotHashes.toList.diff(deSerCheckpointHashes.toList)
+                    checkpointsToGet.traverse {
+                      dao.checkpointService.fullData(_)
+                    }.flatMap { cbs =>
+                      val infoSer = info.copy(acceptedCBSinceSnapshotCache = cbs.flatten).toSnapshotInfoSer()
+                      val res = dao.snapshotService.recentSnapshotInfo
+                        .put(address, infoSer)
+                        .flatTap(_ => IO { logger.warn(s"snapshot/info recentSnapshotInfo updating") })
+                        .map(_ => Array.empty[Byte].some)
+                      val testPut = dao.snapshotService.recentSnapshotInfo.lookup(address)
+                      val check = (res >> testPut).unsafeRunSync()
+                      val didSave = (check.get == infoSer)
+                      logger.debug(s"didSave ${didSave} - for address $address with snapshotHashes ${info
+                        .copy(acceptedCBSinceSnapshotCache = cbs.flatten)
+                        .snapshotHashes} - for node ${dao.id.address}")
+                      res
+                    }
+                  }
+                  APIDirective.handle {
+                    dao.cluster.getNodeState
+                      .map(NodeState.canActAsRedownloadSource)
+                      .ifM(getInfo, IO.pure(None))
+                  } {
+                    case None => complete(StatusCodes.NotFound)
+                    case _    => complete(StatusCodes.OK, Array())
+                  }
+                } else {
+                  val heightPrefix = SnapshotService.snapshotInfoFileHeightPrefix(height)
+                  val snapshotInfoDirectories = dao.snapshotInfoPath.collectChildren(_.isDirectory).toList //need to call toList here
+                  val infoSerOpt = snapshotInfoDirectories.find { d =>
+                    val snapshotInfoDir = d.pathAsString.split("/").reverse.head.startsWith(heightPrefix)
+                    logger.debug(
+                      s"snapshot/info/${height} from node address: ${address} - snapshotInfoDir - ${snapshotInfoDir} - heightPrefix ${heightPrefix} - snapshotInfoDirectories ${snapshotInfoDirectories}"
+                    )
+                    snapshotInfoDir
+                  }.map { f =>
+                    logger.debug(
+                      s"snapshot/info/${height} from node address: ${address} - foundDir - ${f.pathAsString} - heightPrefix ${heightPrefix} - snapshotInfoDirectories ${snapshotInfoDirectories}"
+                    )
+                    RollbackLoader.loadSnapshotInfoSer(f.pathAsString) //todo, filter checkpoint cache with curCheckpointHashes as above ?
```

Comment:
Why `RollbackLoader` is used to GET snapshot info? Please keep single responsibility

Reply:
Just testing